cap-primitives/src/yanix/linux/fs/open.rs

```diff
@@ -35,57 +35,81 @@ struct OpenHow {
 }
 const SIZEOF_OPEN_HOW: usize = std::mem::size_of::<OpenHow>();
 
-/// Call the `openat2` system call.
-fn open_openat2(start: &fs::File, path: &Path, options: &OpenOptions) -> io::Result<fs::File> {
-    let oflags = compute_oflags(options);
-    let mode = options.ext.mode;
+/// Call the `openat2` system call. If the syscall is unavailable, mark it so for future
+/// calls, and fallback to `open_manually_wrapper`
+fn openat2_or_open_manually(
+    start: &fs::File,
+    path: &Path,
+    options: &OpenOptions,
+) -> io::Result<fs::File> {
+    static INVALID: AtomicBool = AtomicBool::new(false);
+    if !INVALID.load(SeqCst) {
```

Comment:
Can we use `Relaxed` instead of `SeqCst` here and in the `store`? We don't care about ordering here, and this is consistent with what [libstd does](https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/rand.rs#L45) in various places.